src/local/BLELocalDevice.cpp

```diff
@@ -51,6 +51,18 @@ int BLELocalDevice::begin()
     return 0;
   }
 
+  uint8_t randomNumber[8];
+  if (HCI.leRand(randomNumber) != 0) {
+    end();
+    return 0;
+  }
+
+  randomNumber[5] |= 0xC0; // Force both MSB bit to b00 in order to define Random Address
```

Comment:
Maybe we can group code from line 55 to line 60 in a dedicated API to generate a random address?

Reply:
@cparata Thanks for the review,
I just try your suggestion,
 but this add too much complexity (for minor added value):
* the dedicated API needs to return a status that needs to be check (so that begin() returns immediately in case of error)
* leRand() get 8 bytes whereas random address need only 6 Bytes, this would add an extra memcopy (it would be ambiguous to declare Ramdom address with 8 bytes)

Reply:
Another comment is about the format of the address. The most two significant bits must be 1 for random static address like described [here](https://devzone.nordicsemi.com/f/nordic-q-a/23456/to-generate-random-address). I don't know if in your computation, you take care about it.

Reply:
This is exactly what is done:
   _* Force both MSB bits to b00 in order to define Static Random Address
   */
  randomNumber[5] |= 0xC0;_

Reply:
Ok good!